zulipterminal/ui_tools/buttons.py

```diff
@@ -370,6 +370,11 @@ def handle_link(self, *_: Any) -> None:
         server_url = self.model.server_url
         if self.link.startswith(urljoin(server_url, "/#narrow/")):
             self.handle_narrow_link()
+        elif self.link.startswith(urljoin(server_url, "/user_uploads/")):
+            # Exit pop-up promptly, let the media download in the background.
+            if isinstance(self.controller.loop.widget, urwid.Overlay):
```

Comment:
We should probably refactor this to be a controller method - we already have one case of this style.